pkg/controller/util/node/controller_utils.go

```diff
@@ -99,13 +101,29 @@ func DeletePods(ctx context.Context, kubeClient clientset.Interface, pods []*v1.
 // pod status, updates it in the apiserver, and returns an error if it
 // encounters one.
 func SetPodTerminationReason(ctx context.Context, kubeClient clientset.Interface, pod *v1.Pod, nodeName string) (*v1.Pod, error) {
-	if pod.Status.Reason == nodepkg.NodeUnreachablePodReason {
+	message := fmt.Sprintf(nodepkg.NodeUnreachablePodMessage, nodeName, pod.Name)
+	updated := false
+	if feature.DefaultFeatureGate.Enabled(features.JobBackoffPolicy) {
+		updated = utilpod.UpdatePodCondition(&pod.Status, &v1.PodCondition{
+			Type:    v1.DeletedFromUnusableNode,
+			Status:  v1.ConditionTrue,
+			Reason:  nodepkg.NodeUnreachablePodReason,
+			Message: message,
+		})
+		if pod.Status.Phase != v1.PodFailed {
+			pod.Status.Phase = v1.PodFailed
+			updated = true
+		}
+	}
+	if pod.Status.Reason != nodepkg.NodeUnreachablePodReason || pod.Status.Message != message {
```

Comment:
In the past, we only checked for the reason to have changed. I think what you have is technically the same, because we are not changing the message, but just to be safe, keep it as it was.

Reply:
ok, updated